snapcraft/internal/elf.py

```diff
@@ -272,6 +287,22 @@ def _get_system_libs() -> FrozenSet[str]:
     return _libraries
 
 
+@lru_cache()
+def _get_magic(path: str) -> str:
+    ms = magic.open(magic.NONE)
+    if ms.load() != 0:
+        raise RuntimeError('Cannot load magic header detection')
+
+    fs_encoding = sys.getfilesystemencoding()
+    path_b = path.encode(
+        fs_encoding, errors='surrogateescape')  # type: bytes
+    return ms.file(path_b)
+
+
+def _is_elf(file_m: str) -> bool:
+    return file_m.startswith('ELF') and 'dynamically linked' in file_m
```

Comment:
This strikes me as wrong. I read the function name as "this is an elf file" but it also checks that it's dynamically linked... and it's a bit unexpected to pass a string rather than a filename.
In both cases where it's used right now we want to know if a file is a dynamic elf library. If that's all we want here, I'd find a single function like `_file_is_dynamically_linked_elf` that takes the path a lot clearer. Thoughts?

Reply:
this means we would need to read the magic hdr twice as we also need the magic header.

Reply:
Fair point. I guess I could live with `_is_dynamically_linked_elf` taking the magic...maybe `..._header` as a suffix? Even better would be having a class wrapping it, but that seems like overkill here (unless you feel like adding that).